src/app/features/charts/components/median-barchart/median-barchart.component.spec.ts

```diff
@@ -0,0 +1,222 @@
+    medianlogcpm: 0.176348063122062,
+    meanlogcpm: -0.323038107200895,
+    quartile3logcpm: 0.391874711168331,
+    maximumlogcpm: 1.9113258251877,
+    tissue: TISSUES[2],
+  },
+];
+
+describe('Component: BarChart - Median', () => {
+  let fixture: ComponentFixture<MedianBarChartComponent>;
+  let component: MedianBarChartComponent;
+  let element: HTMLElement;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [MedianBarChartComponent],
+      imports: [RouterTestingModule],
+      providers: [HelperService],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(MedianBarChartComponent);
+    component = fixture.componentInstance;
+  });
+
+  const setUp = (
+    data: MedianExpression[] = FULL_MOCK_DATA,
+    xAxisLabel: string = XAXIS_LABEL,
+    yAxisLabel: string = YAXIS_LABEL
+  ) => {
+    component.data = data;
+    component.xAxisLabel = xAxisLabel;
+    component.yAxisLabel = yAxisLabel;
+
+    fixture.detectChanges();
+    element = fixture.nativeElement;
+    const chart = element.querySelector('svg g');
+    return { chart };
+  };
+
+  it('should create', () => {
+    setUp();
+    expect(component).toBeTruthy();
+  });
+
+  it('should not render the chart if there is no data', () => {
+    const { chart } = setUp([]);
+
+    expect(component.data?.length).toEqual(0);
+    expect(element.querySelector('.chart-no-data')).toBeTruthy();
+
+    expect(chart).toBeFalsy();
+  });
+
+  it('should not render the chart if all values are negative', () => {
+    const { chart } = setUp(
+      SMALL_MOCK_DATA.map((obj) => {
+        return { ...obj, medianlogcpm: -1 * obj.medianlogcpm };
+      })
+    );
+
+    expect(element.querySelector('.chart-no-data')).toBeTruthy();
+
+    expect(chart).toBeFalsy();
+  });
+
+  it('should render the chart if there is positive data', () => {
+    const createChartSpy = spyOn(component, 'createChart').and.callThrough();
+    const { chart } = setUp();
+
+    expect(component.data?.length).toEqual(FULL_MOCK_DATA.length);
+    expect(createChartSpy).toHaveBeenCalled();
+    expect(chart).toBeTruthy();
+
+    expect(element.querySelector('.chart-no-data')).toBeFalsy();
+  });
+
+  it('should render the chart axes', () => {
+    const { chart } = setUp();
+
+    expect(chart?.querySelector('.x-axis')).toBeTruthy();
+    expect(chart?.querySelector('.y-axis')).toBeTruthy();
+  });
+
+  it('should render the correct number of bars', () => {
+    const { chart } = setUp();
+
+    expect(chart?.querySelectorAll('rect').length).toEqual(
+      FULL_MOCK_DATA.length
+    );
+  });
+
+  it('should not render bars for negative values', () => {
+    const { chart } = setUp([
+      { ...SMALL_MOCK_DATA[0], medianlogcpm: -0.01 },
+      ...SMALL_MOCK_DATA.slice(1),
+    ]);
+
+    expect(chart?.querySelectorAll('rect').length).toEqual(
+      SMALL_MOCK_DATA.length - 1
+    );
+  });
```

Comment:
Ensure that new implementation matches the current behavior, i.e. that negative medianlogcpm values are removed from the chart